homeassistant/components/vesync/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""Binary Sensor for VeSync."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from pyvesync.vesyncbasedevice import VeSyncBaseDevice
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .common import rgetattr
+from .const import DOMAIN, VS_FANS
+from .entity import VeSyncBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class VeSyncBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes custom binary sensor entities."""
+
+    is_on: Callable[[VeSyncBaseDevice], bool] | None = None
+
+
+SENSOR_DESCRIPTIONS: tuple[VeSyncBinarySensorEntityDescription, ...] = (
+    VeSyncBinarySensorEntityDescription(
+        key="water_lacks",
+        translation_key="water_lacks",
+        is_on=lambda device: device.water_lacks,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    VeSyncBinarySensorEntityDescription(
+        key="details.water_tank_lifted",
+        translation_key="water_tank_lifted",
+        is_on=lambda device: device.details["water_tank_lifted"],
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up binary_sensor platform."""
+    entities: list[VeSyncBinarySensor] = []
+    for device in hass.data[DOMAIN][VS_FANS]:
+        for description in SENSOR_DESCRIPTIONS:
+            if rgetattr(device, description.key) is not None:
+                entities.append(VeSyncBinarySensor(device, description))  # noqa: PERF401
```

Comment:
Why the noqa? This can be a fine list comprehension

Reply:
It throws an error asking me to switch to extend.  When I tried extend it fails tests.

Reply:
![alt text](image.png)  It is only warning but I am trying to address these as I go.